# Problema 1

Considere la sucesión $a_{n} = 3n^{2} + 7$. Por medio de un código en *Python* despliega en pantalla:

1. Los primeros $5$ términos.
2. Los $5$ términos que vienen inmediatamente después del decimoquinto término.

In [5]:
1.
print("Los primero 5 terminos de la sucesión")

def a_n(n):
    return 3*n**2+7

for n in range(1,6):
    termino=a_n(n)
    print(f"a_{n}=3*{n}^2+7={termino}")

Los primero 5 terminos de la sucesión
a_1=3*1^2+7=10
a_2=3*2^2+7=19
a_3=3*3^2+7=34
a_4=3*4^2+7=55
a_5=3*5^2+7=82


In [6]:
2.
print(''''Los primeros 5 terminos que vienen inmediatamente después del decimo quinto termino,
      es decir los terminos 16,17,18,19,20''')

def a_n(n):
    return 3*n**2+7

for n in range(16,21):
    termino=a_n(n)
    print(f"a_{n}=3*{n}^2+7={termino}")

'Los primeros 5 terminos que vienen inmediatamente después del decimo quinto termino,
      es decir los terminos 16,17,18,19,20
a_16=3*16^2+7=775
a_17=3*17^2+7=874
a_18=3*18^2+7=979
a_19=3*19^2+7=1090
a_20=3*20^2+7=1207


# Problema 2

Considere la sucesión $g_{n} = 5n^{3}$. Implemente un código en *Python* que muestre en pantalla:

1. Los primeros $4$ términos de la sucesión.
2. Los $4$ términos que vienen inmediatamente después del octavo término.
3. Determinar si el término $40.000$ pertenece a la sucesión. Si es así, indicar en qué posición se encuentra.

In [4]:
1.
print("Los primeros 4 terminos de la sucesión:")

def g_n(n):
    return 5*n**3

for n in range(1,5):
    termino=g_n(n)
    print(f"a_{n}=5*{n}^3={termino}")

Los primeros 4 terminos de la sucesión:
a_1=5*1^3=5
a_2=5*2^3=40
a_3=5*3^3=135
a_4=5*4^3=320


In [5]:
2
print('''Los 4 terminos que vienen inmediatamente después del octavo termino,
      es decir los terminos 9,10,11,12''')

def g_n(n):
    return 5*n**3

for n in range(9,13):
    termino=g_n(n)
    print(f"a_{(n)}=5{n}^3={termino}")

Los 4 terminos que vienen inmediatamente después del octavo termino,
      es decir los terminos 9,10,11,12
a_9=59^3=3645
a_10=510^3=5000
a_11=511^3=6655
a_12=512^3=8640


In [ ]:
3.
print("Determinar si el término 40.000 pertenece a la sucesión. Si es así, indicar en qué posición se encuentra.")

n_calculado=(40000/5)**(1/3) 

print(n_calculado)

print("40000 si pertenece a la sucesion y está en la posición 20")

Determinar si el término 40.000 pertenece a la sucesión. Si es así, indicar en qué posición se encuentra.
19.999999999999996
40000 si pertenece a la sucesion y está en la posición 20


# Problema 3

La sucesión de Fibonacci es una serie de números en la que cada número, desde el tercero en adelante, es la suma de los dos anteriores. Comienza con el $0$ y $1$, tal como se observa a continuación:

$$0, 1, 1, 2, 3, 5, 8, 13, ...$$

Formalmente, la sucesión se define de la siguiente manera:

\begin{split}
F(0) &= 0\\
F(1) &= 1\\
F(n) &= F(n - 1) + F(n - 2), \ \text{ para $n \geq 2$}
\end{split}

Esta sucesión se utiliza en una variedad de contextos en informática y algoritmos debido a sus propiedades matemáticas y patrones.

1. Implementa un código en *Python* que muestre en pantalla los primeros $20$ términos de la sucesión.
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $30$ términos de la sucesión de Fibonacci||.

In [11]:
1.
print("Los primeros 20 terminos de la sucesión:")

def f(n):
    if n==0:
        return 0
    elif n==1:
        return 1
    elif n>=2:
        return f(n-1)+f(n-2)

print("Los primeros 20 de Fibonacci son:")
for n in range (0,21):
    termino=f(n)
    print(F"{n+1}) {n} = {termino}")


Los primeros 20 terminos de la sucesión:
Los primeros 20 de Fibonacci son:
1) 0 = 0
2) 1 = 1
3) 2 = 1
4) 3 = 2
5) 4 = 3
6) 5 = 5
7) 6 = 8
8) 7 = 13
9) 8 = 21
10) 9 = 34
11) 10 = 55
12) 11 = 89
13) 12 = 144
14) 13 = 233
15) 14 = 377
16) 15 = 610
17) 16 = 987
18) 17 = 1597
19) 18 = 2584
20) 19 = 4181
21) 20 = 6765


In [15]:
2.
print("la suma de los primeros 30 términos de la sucesión de Fibonacci es: ")

def f(n):
    if n==0:
        return 0
    elif n==1:
        return 1
    elif n>=2:
        return f(n-1)+f(n-2)

lista=[]

for n in range(0,31):
    termino=f(n)
    lista.append(termino)

suma=sum(lista)
print(suma)

la suma de los primeros 30 términos de la sucesión de Fibonacci es: 
2178308


# Problema 4

Implementa un código que permita guardar en una lista la sucesión de los primeros $n$ números pares, con $n \geq 1$. Luego, determina:

1. ¿Cuál es el valor del décimo término?
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $100$ números pares.
3. ¿En qué posición se encuentra el número $58$ dentro de la sucesión?

In [8]:
1.

lista=[]

def f(n):
    if n>=1:
        return 2*n
    
for n in range(1,11):
    termino=f(n)
    lista.append(termino)

print(lista)
print(f"El decimo termino es: {lista[9]}")

# for n in range(0,9):
#     print(lista[n])

    

[2, 4, 6, 8, 10, 12, 14, 16, 18, 20]
El decimo termino es: 20


In [16]:
2.
lista=[]

def f(n):
    if n>=1:
        return 2*n
    
for n in range(1,101):
    termino=f(n)
    lista.append(termino)

suma=sum(lista)

print(lista)
print(f"La suma de los primeros 100 numeros pares es: {suma}")



[2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120, 122, 124, 126, 128, 130, 132, 134, 136, 138, 140, 142, 144, 146, 148, 150, 152, 154, 156, 158, 160, 162, 164, 166, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 188, 190, 192, 194, 196, 198, 200]
La suma de los primeros 100 numeros pares es: 10100


In [19]:
3.
lista=[]

def f(n):
    if n>=1:
        return 2*n
    
for n in range(1,101):
    termino=f(n)
    lista.append(termino)

posicion=lista.index(58)

print(f"El numero 58 se encuentra en la posición: {posicion}")

El numero 58 se encuentra en la posición: 28


# Problema 5

Un nuevo software tiene $500$ usuarios y se espera que el número de usuarios crezca un $15\%$ respecto al mes anterior. Considere que $b_{n}$ es el número de usuarios transcurridos $n$ meses.

1. Calcula cuántos usuarios tendrá el software al cabo de tres meses. ¿Y al cabo de seis meses?
2. Mediante un código en *Pyhton* muestra la cantidad de usuarios que tendrá el software cada mes durante el primer año.
2. ¿Cuántos usuarios tendrá en total el software luego de un año? Utilice ciclos *for* para calcular.

In [23]:
1.
def b_n(n):
    return 500*(1.15)**n

termino1=b_n(3)
termino2=b_n(6)

print(f"El numero de usuarios que tendrá el software al cabo de tres meses es: {termino1:.1f}")
print(f"El numero de usuarios que tendrá el software al cabo de seis meses es: {termino2:.1f}")

El numero de usuarios que tendrá el software al cabo de tres meses es: 760.4
El numero de usuarios que tendrá el software al cabo de seis meses es: 1156.5


In [31]:
# 2. INCOMPLETO !!!
lista=[]

def b_n(n):
    return 500*(1.15)**n

for n in range(1,13):
    termino=b_n(n)
    lista.append(termino)

print(f"La cantidad de usuarios que tendrá el software cada mes durante el primer año son: {lista}")

La cantidad de usuarios que tendrá el software cada mes durante el primer año son: [575.0, 661.2499999999999, 760.4374999999999, 874.5031249999997, 1005.6785937499997, 1156.5303828124995, 1330.0099402343744, 1529.5114312695305, 1758.9381459599597, 2022.7788678539534, 2326.195698032046, 2675.125052736853]


In [37]:
3.
lista=[]

def b_n(n):
    return 500*(1.15)**n

for n in range(1,13):
    termino=b_n(n)
    lista.append(termino)

suma=sum(lista)

print(f"Luego de un año el software tendrá {suma:.2f} usuarios")

Luego de un año el software tendrá 16675.96 usuarios


# Problema 6

Un estudiante de ingeniería informática planea ahorrar para un proyecto personal. Para esto, el primer mes deposita en una cuenta $\$12.000$, el segundo mes deposita $\$14.000$ y así sucesivamente, de modo que cada mes aumenta el depósito en $\$2.000$. Considere que $a_{i}$ corresponde al depósito del estudiante en el mes $i$.

1. Implementa un código que permita mostrar en pantalla lo que depositó mensualmente durante el primer año.
2. ¿Cuánto depositó el estudiante en febrero del segundo año?
3. ¿Cuál será el total ahorrado al finalizar dos años?

In [8]:
1.
def a_i(n):
    return 12000+2000*n

total_año=0
for n in range(0,13):
    mensual=a_i(n)
    total_año+=mensual
    print(f"Mes {n+1}= {mensual}")

print(f"El total de lo que depositó en un año fue: {total_año}")

Mes 1= 12000
Mes 2= 14000
Mes 3= 16000
Mes 4= 18000
Mes 5= 20000
Mes 6= 22000
Mes 7= 24000
Mes 8= 26000
Mes 9= 28000
Mes 10= 30000
Mes 11= 32000
Mes 12= 34000
Mes 13= 36000
El total de lo que depositó en un año fue: 312000


In [9]:
2.
def a_i(n):
    return 12000+2000*n

febrero_2ndo_año=a_i(14)

print(f"En Febrero del segundo año el estudiante depositó: {febrero_2ndo_año}")

En Febrero del segundo año el estudiante depositó: 40000


In [11]:
3.
def a_i(n):
    return 12000+2000*n

total_año=0
for n in range(0,24):
    mensual=a_i(n)
    total_año+=mensual
    # print(f"Mes {n+1}= {mensual}")

print(f"El total de lo que depositó en en 2 años es: {total_año}")

El total de lo que depositó en en 2 años es: 840000


# Problema 7

Por lo general, los proyectos son organizados en varias etapas que deben completarse secuencialmente.

La fase inicial de un proyecto de software tiene una duración de $3$ meses y cada fase subsecuente reduce su duración en un $10 \%$ debido a la experiencia ganada. Determine:

1. ¿Cuánto dura la quinta fase?
2. Calcula la duración total del proyecto después de $10$ fases.

In [22]:
def a_i(n):
    return 90*(0.9**n)

quinta_fase=a_i(4)
decima_fase=a_i(9)

print(f"La quinta fase dura: {quinta_fase:.1f}")
print(f"La decima fase dura: {decima_fase:.1f}")

La quinta fase dura: 59.0
La decima fase dura: 34.9


# Problema 8

En una sucesión aritmética, la diferencia entre un término y el anterior es $4$ y el décimo término es $-20$. Determine:

1. El primer término de la sucesión.
2. La expresión del término de lugar $n$.
3. Utilice un código en *Python* para calcular el término que ocupa el lugar $100$.
4. Determinar la posición del número $1.680$ en la secuencia. Calcule utilizando un ciclo *while*.

# Problema 9

El tercer término de una sucesión geométrica es $5$ y el sexto término es $40$. Determine:

1. La razón geométrica de la sucesión.
2. El primer término de la sucesión.
3. La expresión del término de lugar $n$.
4. Utilice un código en *Python* para calcular el término que ocupa el lugar $8$.
4. ¿Qué lugar ocupa en esta sucesión el término $20.480$? Calcule utilizando un ciclo *while*.

# Problema 10

Una empresa de tecnología está optimizando sus servidores. Para ello mide las transacciones por segundo (TPS) de sus servidores en cada mes. Se ha obtenido que en el segundo mes, el rendimiento del servidor fue $1,2$ miles de TPS y en el quinto mes, el rendimiento fue $2,7$ miles de TPS. Si el número de TPS que hay mes a mes sigue una sucesión geométrica, determina utilizando un código en *Python*:

1. ¿Cuál es el rendimiento el noveno mes?
2. ¿En qué mes el rendimiento es de $23,4$ miles de TPS?

Utilice dos decimales para el cálculo.

# Problema 11

Una aplicación de redes sociales se estrenó el $1$ de enero con una cierta cantidad de usuarios. A partir de ahí, el número de usuarios nuevos de cada día fue siempre el mismo. El $3$ de enero tenía $500$ usuarios en total y el $10$ de enero tenía $1.501$ usuarios totales.Determine:

1. ¿Con cuántos usuarios comenzó a funcionar la aplicación?
2. ¿Cuántos usuarios tendrá la aplicación el $31$ de enero?
3. Si la aplicación tiene $5.505$ usuarios, ¿cuántos meses lleva funcionando?



# Problema 12

Una empresa de publicidad ofrece para un puesto de trabajo un salario inicial y un aumento mensual en una razón constante. Javier se presenta a la entrevista, y el encargado del departamento de recursos humanos le explica que en el segundo mes de trabajo su sueldo será de $\$402.000$, y que en el tercer mes de trabajo será de $\$404.010$. Determine:

1. El salario aproximado que Javier recibiría en el décimo mes de trabajo en la empresa.
2. ¿Cuál sería el ingreso total recibido por los sueldos de Javier durante el tercer año de trabajo?

# Problema 13

Una empresa de ciberseguridad capacita trabajadores cada trimestre. En el primer trimestre capacitaron a $30$ trabajadores, en el segundo trimestre a $60$ trabajadores, y en el tercer trimestre a $120$ trabajadores. Si en los siguientes trimestres la empresa mantiene el mismo patrón de crecimiento, determine utilizando *Python*:

1. ¿A cuántos trabajadores capacitarán en el octavo trimestre?
2. Si la empresa lleva en el mercado $3$ años, ¿a cuántos trabajadores han capacitado en total?

# Problema 14

Al llenar un tambor con agua, se registra que durante el minuto $4$ el tambor recibe $128$ litros, y durante el minuto $9$ recibe $88$ litros. Sabiendo que en cada minuto recibe una cantidad $d$ de litros de agua menos que en el minuto anterior, determine:

1. ¿En qué instante el tambor deja de recibir agua?
2. ¿Cuántos litros recibe en total el tambor?

Realice los cálculos anteriores utilizando *Python*.